browser/ui/webui/new_tab_page/new_tab_page_branded_view_counter.cc

```diff
@@ -59,8 +80,12 @@ class NewTabPageBrandedViewCounterFactory
   // BrowserContextKeyedServiceFactory:
   KeyedService* BuildServiceInstanceFor(
       content::BrowserContext* browser_context) const override {
-    return new NewTabPageBrandedViewCounter(
+    NewTabPageBrandedViewCounter* instance = new NewTabPageBrandedViewCounter(
         Profile::FromBrowserContext(browser_context));
+    NTPSponsoredImagesComponentManager* manager =
+        g_brave_browser_process->ntp_sponsored_images_component_manager();
+    manager->AddObserver(instance);
```

Comment:
There should be corresponding `RemoveObserver` calling.

Reply:
I believe this is now done